Lib/test/test_queue.py

```diff
@@ -317,38 +317,39 @@ def test_shutdown_all_methods_in_one_thread(self):
     def test_shutdown_immediate_all_methods_in_one_thread(self):
         return self._shutdown_all_methods_in_one_thread(True)
 
-    def _write_msg_thread(self, q, n, results, delay,
-                            i_when_exec_shutdown,
-                            event_start, event_end):
-        event_start.wait()
+    def _write_msg_thread(self, q, n, results,
+                            i_when_exec_shutdown, event_shutdown,
+                            event_start):
+        put_atleast = i_when_exec_shutdown//2
         for i in range(1, n+1):
             try:
                 q.put((i, "YDLO"))
-                results.append(True)
             except self.queue.ShutDown:
                 results.append(False)
-            # triggers shutdown of queue
+                break
+
+            # Be sure that all write_threads
+            # put few items into the queue.
+            if i == put_atleast:
+                event_start.wait()
+
+            # Triggers shutdown of queue.
             if i == i_when_exec_shutdown:
-                event_end.set()
-            time.sleep(delay)
-        # end of all puts
+                if not event_shutdown.is_set():
```

Comment:
Sorry if I missed something. Presumably, we don't need to check since the loop can only be here once?

Reply:
At this point, several `_write_msg_threads` threads may together reach the `i_when-exec_shutdown` threshold. The test consists in ensuring that only one of them has triggered the `event_shutdown` (And wake up the `_shutdown_thread` thread).

Reply:
I see, that makes sense. Thanks for the explanation